napari/layers/base/base.py

```diff
@@ -610,6 +610,7 @@ def get_message(self):
 
         value = self._value
 
+        # FIXME: deprecation warning... but logic unclear
```

Comment:
yeah, this isn't so nice. Some layers return value as a singleton, hence the `is not None`, but some layers like `Shapes` can return a tuple, I think the following would work

```python
if value is not None and not np.all([v == None for v in value])
```

Reply:
actually, looks like `value` can be: `None`, a scalar, a `tuple`, and even a`tuple` with a nested numpy `array` (for rgb_pyramid).   `for v in value` will throw an error if `value` is not iterable, so that won't work.  Do we just want to check whether it's a non-null scalar, or an iterable with no null values?  (or an iterable with at least one non-null value?)

Reply:
that sounds reasonable for now, then we might want to refactor the layers so the output of value is a little more constrained